### PR TITLE
Ensure Party Map always opens rooms in a new tab

### PR DESCRIPTION
**Pull request by 0xdevalias**

### CODE REVIEW

File: src/components/templates/Camp/components/MapRoomOverlay.tsx [hunk 1]
```diff
@@ -196,14 +193,14 @@ export const MapRoomOverlay: React.FC<MapRoomOverlayProps> = ({
...
               {/* TODO: extract this into a generalised link component that handles internal/external correctly? */}
-              <a
+              <button
                 className="btn btn-block btn-small btn-primary"
-                onClick={onClick3}
-                href={currentRoomUrl}
-                {...getExtraLinkProps(isRoomExternal)}
+                onClick={onJoinRoom}
+                // href={currentRoomUrl}
+                // {...getExtraLinkProps(isRoomExternal)}
```
Comment (thecadams) [inline src/components/templates/Camp/components/MapRoomOverlay.tsx:195]: Looks like invalid JSX to me.

File: src/components/templates/Camp/components/MapRoomOverlay.tsx [hunk 2]
```diff
@@ -97,23 +97,20 @@ export const MapRoomOverlay: React.FC<MapRoomOverlayProps> = ({
...
 
-      if (isExternalUrl(room.url)) {
-        openRoomUrl(room.url);
-      } else {
-        window.location.href = getRoomUrl(room.url);
-      }
-
       enterCampRoom(room);
+      // openRoomUrl(room.url);
+      // @debt use the helpers properly, we want this to always open in a new url
```
Comment (thecadams) [inline src/components/templates/Camp/components/MapRoomOverlay.tsx:97]: :+1:

File: src/components/templates/Camp/components/MapRoomOverlay.tsx [hunk 3]
```diff
@@ -97,23 +97,20 @@ export const MapRoomOverlay: React.FC<MapRoomOverlayProps> = ({
     (e) => {
       e.stopPropagation();
 
-      if (isExternalUrl(room.url)) {
-        openRoomUrl(room.url);
-      } else {
-        window.location.href = getRoomUrl(room.url);
-      }
-
```
Comment (thecadams) [inline src/components/templates/Camp/components/MapRoomOverlay.tsx:105]: nit: commented code

File: src/components/templates/Camp/components/MapRoomOverlay.tsx [hunk 4]
```diff
@@ -42,8 +42,8 @@ export const MapRoomOverlay: React.FC<MapRoomOverlayProps> = ({
   const isSelectedRoom = room === selectedRoom;
   const hasAttendance = attendances[`${venue.name}/${room.title}`];
 
-  const isRoomExternal = isExternalUrl(room.url);
-  const currentRoomUrl = getRoomUrl(room.url);
+  // const isRoomExternal = isExternalUrl(room.url);
+  // const currentRoomUrl = getRoomUrl(room.url);
```
Comment (thecadams) [inline src/components/templates/Camp/components/MapRoomOverlay.tsx:41]: nit: commented code

File: src/components/templates/Camp/components/MapRoomOverlay.tsx [hunk 5]
```diff
@@ -97,23 +97,20 @@ export const MapRoomOverlay: React.FC<MapRoomOverlayProps> = ({
...
   // TODO: should this also openRoomUrl as above? If so, let's use the same function for it.. if not, why?
   // TODO: what is a better semantic name for this?
-  const onClick3 = useCallback(() => enterCampRoom(room), [
-    enterCampRoom,
-    room,
-  ]);
+  // const onClick3 = useCallback(() => enterCampRoom(room), [
+  //   enterCampRoom,
+  //   room,
+  // ]);
```
Comment (thecadams) [inline src/components/templates/Camp/components/MapRoomOverlay.tsx:108]: nit: commented code

File: src/components/templates/Camp/components/MapRoomOverlay.tsx [hunk 6]
```diff
@@ -196,14 +193,14 @@ export const MapRoomOverlay: React.FC<MapRoomOverlayProps> = ({
 
             <div className="camp-venue-actions">
               {/* TODO: extract this into a generalised link component that handles internal/external correctly? */}
-              <a
+              <button
```
Comment (thecadams) [inline src/components/templates/Camp/components/MapRoomOverlay.tsx:191]: Good to see this become a `button`.

**Review by thecadams — COMMENTED** [review]
Comments on this version, but I imagine there are no surprises in here.

File: src/components/templates/Camp/components/MapRoomOverlay.tsx [hunk 7]
```diff
@@ -97,23 +92,20 @@ export const MapRoomOverlay: React.FC<MapRoomOverlayProps> = ({
...
       e.stopPropagation();
 
-      if (isExternalUrl(room.url)) {
-        openRoomUrl(room.url);
-      } else {
-        window.location.href = getRoomUrl(room.url);
-      }
-
       enterCampRoom(room);
+      // openRoomUrl(room.url);
```
Comment (thecadams) [inline src/components/templates/Camp/components/MapRoomOverlay.tsx:96]: nit: commented code

File: src/components/templates/Camp/components/MapRoomOverlay.tsx (hunk 1 above)
Comment (0xdevalias) [reply src/components/templates/Camp/components/MapRoomOverlay.tsx:195]: It's valid to comment code within a component tag in JSX like this.

File: src/components/templates/Camp/components/MapRoomOverlay.tsx (hunk 3 above)
Comment (0xdevalias) [reply src/components/templates/Camp/components/MapRoomOverlay.tsx:105]: It leaves context to the speedy hack being done here. And all of this is 🔥  in https://github.com/sparkletown/sparkle/pull/840 refactor anyway.

**Comment by 0xdevalias** [comment]
🔨  glass

**Pull request merged**